subliminal/providers/itasa.py

```diff
@@ -0,0 +1,485 @@
+        :return: the show id, if found.
+        :rtype: int or None
+
+        """
+        series_sanitized = sanitize(series).lower()
+        show_ids = self._get_show_ids()
+        show_id = None
+
+        # attempt with country
+        if not show_id and country_code:
+            logger.debug('Getting show id with country')
+            show_id = show_ids.get('%s %s' % (series_sanitized, country_code.lower()))
+
+        # attempt clean
+        if not show_id:
+            logger.debug('Getting show id')
+            show_id = show_ids.get(series_sanitized)
+
+        # search as last resort
+        if not show_id:
+            logger.warning('Series not found in show ids')
+            show_id = self._search_show_id(series)
+
+        return show_id
+
+    @region.cache_on_arguments(expiration_time=EPISODE_EXPIRATION_TIME)
+    def _download_zip(self, sub_id):
+        # download the subtitle
+        logger.info('Downloading subtitle %r', sub_id)
+
+        params = {
+            'authcode': self.auth_code,
+            'apikey': self.apikey,
+            'subtitle_id': sub_id
+        }
+
+        r = self.session.get(self.server_url + 'subtitles/download', params=params, timeout=30)
+        r.raise_for_status()
+
+        return r.content
+
+    def _get_season_subtitles(self, show_id, season, sub_format):
+        params = {
+            'apikey': self.apikey,
+            'show_id': show_id,
+            'q': 'Stagione %%%d' % season,
+            'version': sub_format
+        }
+        r = self.session.get(self.server_url + 'subtitles/search', params=params, timeout=30)
+        r.raise_for_status()
+        root = etree.fromstring(r.content)
+
+        if int(root.find('data/count').text) == 0:
+            logger.warning('Subtitles for season not found, try with rip suffix')
+
+            params['version'] = sub_format + 'rip'
+            r = self.session.get(self.server_url + 'subtitles/search', params=params, timeout=30)
+            r.raise_for_status()
+            root = etree.fromstring(r.content)
+            if int(root.find('data/count').text) == 0:
+                logger.warning('Subtitles for season not found')
+                return []
+
+        subs = []
+        # Looking for subtitles in first page
+        season_re = re.compile('.*?stagione 0*?%d.*' % season)
+        for subtitle in root.findall('data/subtitles/subtitle'):
+            if season_re.match(subtitle.find('name').text.lower()):
+                logger.debug('Found season zip id %d - %r - %r',
+                             int(subtitle.find('id').text),
+                             subtitle.find('name').text,
+                             subtitle.find('version').text)
+
+                content = self._download_zip(int(subtitle.find('id').text))
+                if not is_zipfile(io.BytesIO(content)):   # pragma: no cover
+                    if 'limite di download' in content:
+                        raise TooManyRequests()
```

Comment:
@Elettronik IMHO I would do like this: raise DownloadLimitExceeded('You reached the download limit')``